collect_app/src/main/java/org/odk/collect/android/injection/config/AppDependencyModule.java

```diff
@@ -166,7 +165,7 @@ public ActivityAvailability providesActivityAvailability(Context context) {
 
     @Provides
     InstallIDProvider providesInstallIDProvider(Context context) {
-        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
+        SharedPreferences preferences = context.getSharedPreferences("meta", Context.MODE_PRIVATE);
```

Comment:
I think that this is good but I'm concerned that it will get lost and forgotten. How about marking `KEYS_WE_SHOULD_NOT_RESET` as deprecated and describing this strategy? Maybe `meta` should be a constant somewhere?

Reply:
Yeah I was in two minds about this as I'm not a fan of the existing shared prefs abstractions and the keys classes. I do agree that it would be nice to have something that makes the `meta` prefs more discoverable though so they get used for other things. I'll have a think.

Reply:
As a note I'd like to keep the test with a hardcoded/inline as this prevents someone changing the constant (in the code) and the test still passing. It's checking configuration as well as behaviour.

Reply:
I'm going to go with a provider for the `meta` prefs for now. I didn't want to start building a whole `MetaSharedPrefences` to mimic the other ones. I like that we have abstraction for prefs storage though so I might follow up with a refactor that pulls out the shared logic around shared prefs and maybe improves the interfaces a little.